docs/book/src/reference/completion.md

```diff
@@ -0,0 +1,62 @@
+# Enable shell auto completion
+
+This  guide will cover:
+
+- [Upgrade Bash](#upgrade-bash)
+- [Install bash-completion](#install-bash-completion)
+
+## Prerequisites
+
+- [Homebrew](https://docs.brew.sh/Installation)
+
+## Bash on MacOS
+The kubebuilder completion script for Bash can be generated with kubebuilder completion bash. Sourcing this script in your shell enables kubebuilder completion.
```

Comment:
If you're going to discuss installation on macos, you have to discuss installation on linux/windows too. I'd delegate the task of figuring that out to the user given all the different environments they could be using, and just say "ensure you have bash 4.1+ installed"

Also is there a version requirement for zsh? If so, mention that as well.

Reply:
i was hoping, once this PR is done, someone else who have linux , can contribute to add details on specific steps. WDYT?

Reply:
@mengqiy @DirectXMan12 how do you feel about documenting installation for distros?